app/hooks/useHover.ts

```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import useUnmount from "./useUnmount";
+
+type Props = {
+  /** Denotes the duration to wait until it's considered as a hover event. */
+  duration: number;
+};
+
+/**
+ * Hook to detect if an element is hovered by the user.
+ *
+ * This signals the hover only on the first time an element is hovered.
+ *
+ * @returns {Object} object - The returned object.
+ * @returns {boolean} object.hovered - Signals when an element is hovered by the user.
+ * @returns {function(): void} object.onMouseEnter - Attach this function to the observed element's onMouseEnter event handler.
+ * @returns {function(): void} object.onMouseLeave - Attach this function to the observed element's onMouseLeave event handler.
+ */
+const useHover = ({ duration }: Props) => {
```

Comment:
Would it be a nicer API to pass a `ref` as a prop and handle the event binding internally?

Reply:
Agree - makes it clean.